src/BotwinExtensions.cs

```diff
@@ -82,17 +83,20 @@ private static RequestDelegate CreateRouteHandler(string path, Type moduleType)
 
                 if (module.Before != null)
                 {
+                    logger?.LogTrace("Executing module before hook");
                     shouldContinue = await module.Before(ctx);
                 }
 
                 if (shouldContinue)
                 {
                     // run the route handler
+                    logger?.LogTrace($"Executing module route handler for {ctx.Request.Method} {path}");
```

Comment:
Use structured logging instead of string interpolation:

```
logger.LogTrace("Executing module route handler for {Method} {Path}", ctx.Request.Method, path);
```